src/System.Management.Automation/engine/parser/Parser.cs

```diff
@@ -5671,16 +5677,11 @@ internal Ast CommandRule(bool forDynamicKeyword)
                         case TokenKind.Semi:
                         case TokenKind.AndAnd:
                         case TokenKind.OrOr:
+                        case TokenKind.Ampersand:
```

Comment:
& is a statement separator like in ksh so in your examples, you'd get an array of two objects: the job object and "foo".